mllib/src/main/scala/org/apache/spark/ml/optim/WeightedLeastSquares.scala

```diff
@@ -108,6 +101,57 @@ private[ml] class WeightedLeastSquares(
           "Consider setting fitIntercept=true.")
       }
     }
+    /*
+     * If more than one of the features in the data are constant (i.e. data matrix has constant
+     * columns), then A^T.A is no longer positive definite and Cholesky decomposition fails
+     * (because the normal equation does not have a solution).
+     * In order to find a solution, we need to drop constant columns from the data matrix. Or,
+     * we can drop corresponding column and row from A^T.A matrix.
+     * Once we drop rows/columns from A^T.A matrix, the Cholesky decomposition will produce
+     * correct coefficients. But, for the final result, we need to add zeros to the list of
+     * coefficients corresponding to the constant features.
+     */
+    val aVarRaw = summary.aVar.values
```

Comment:
I assume that aVar will be dense here. If you would like to do it in this way, then `val aVarRaw = summary.aVar.toArray` to make sure it's dense. If `summary.aVar` is already dense, this will just return the pointer to that dense array without copying.